controllers/user.js

```diff
@@ -0,0 +1,343 @@
+                    }
+                });
+            }
+        });
+    }
+}
+
+exports.resetpassword = function(req, res) {
+    var {
+        contact_no,
+        access_token
+    } = req.body;
+    manValues = [contact_no, access_token];
+    checkBlank = commfunc.checkBlank(manValues);
+    if (checkBlank == 1) {
+        responses.parameterMissing(res);
+    } else {
+        var OTP = commfunc.generateRandomString();
+        UserModal.showalldata(access_token, function(showresult) {
+            if (showresult == 0) {
+                responses.sendError(res);
+            } else {
+                if (showresult[0].contact_no == contact_no) {
+                    data = {
+                        OTP: OTP
+                    };
+                    condition = {
+                        access_token: access_token
+                    };
+                    UserModal.updateUserData(data, condition, function(updatedResult) {
+                        if (updatedResult == 0) {
+                            responses.sendError(res);
+                        } else {
+                            body = "your OTP is " + OTP + "valid for 3 minutes";
+                            commfunc.sendotp(contact_no, OTP, body);
+                            responses.success(res, updatedResult)
+                        }
+
+                    });
+
+                }
+
+            }
+
+        });
+
+    }
+}
+exports.updatepassword = function(req, res) {
+    var {
+        OTP,
+        password,
+        access_token
+    } = req.body;
+    manValues = [OTP, password, access_token];
+    var checkBlank = commfunc.checkBlank(manValues);
+    if (checkBlank == 1) {
+        responses.parameterMissing(res);
+    } else {
+        UserModal.showalldata(access_token, function(showresult) {
+            if (showresult == 0) {
+                responses.sendError(res);
+            } else {
+                if (showresult[0].OTP == OTP) {
+                    data = {
+                        password: md5(password),
+                        OTP: 0
+                    };
+                    condition = {
+                        access_token: access_token
+                    };
+                    UserModal.updateUserData(data, condition, function(updatedResult) {
+                        if (updatedResult == 0) {
+                            responses.sendError(res);
+                        } else {
+                            UserModal.showalldata(access_token, function(showresult) {
+                                if (showresult == 0) {
+                                    responses.sendError(res);
+                                } else {
+                                    contact_no = showresult[0].contact_no;
+                                    date = new Date();
+                                    body = "your password has been changed on " + date;
+                                    commfunc.sendotp(contact_no, OTP, body);
```

Comment:
no use of date, why you define data and if date is required then where you define the variable